module1/homework/grayscaleImages/compression.cpp

```diff
@@ -0,0 +1,60 @@
+#include "compression.hpp"
+
+#include <algorithm>
+#include <array>
+#include <iostream>
+#include <utility>
+#include <vector>
+
+std::vector<std::pair<uint8_t, uint8_t>> compressGrayscale(std::array<std::array<uint8_t, width>, height>& bitmap) {
+    std::vector<std::pair<uint8_t, uint8_t>> compressedData;
+
+    for (const auto& row : bitmap) {
```

Comment:
Think about reserve memory in vector. What is maximum value for sure ( tip: every single position is different)